docs/content/guide/component.ngdoc

```diff
@@ -0,0 +1,453 @@
+
+When not to use Components:
+
+- for directives that rely on DOM manipulation, adding event listeners etc, because the compile
+and link functions are unavailable
+- when you need advanced directive defintion options like priority, terminal, multi-element
+
+## Creating and configuring a component
+
+We create an {@link module Angular Module}, `myApp`, for our application. Then we add the component
+config object to the module using the `.component()` method (note that the method does not take a
+factory function, unlike the `directive()` method).
+
+<example name="heroComponentSimple" module="heroApp">
+<file name="heroDetail.js">
+  var myMod = angular.module('heroApp', []);
+
+  function HeroDetailController($scope, $element, $attrs) {
+    // This is called once all bindings are initialized
+    this.$onInit = function() {
+      if (!this.hero.location) {
+        this.hero.location = 'unknown';
+      }
+    };
+  }
+
+  myMod.component('heroDetail', {
+    templateUrl: 'heroDetail.html',
+    controller: HeroDetailController,
+    bindings: {
+      hero: '='
+    }
+  });
+</file>
+<file name="index.html">
+  <!-- components match only elements -->
+  <b>Hero</b><br>
+  <hero-detail hero="{name: 'Spawn'}"></hero-detail>
+</file>
+<file name="heroDetail.html">
+  <span>Name: {{$ctrl.hero.name}}</span><br>
+  <span>Location: {{$ctrl.hero.location}}</span>
+</file>
+</example>
+
+It's also possible to add components via {@link $compileProvider#component} in a module's config phase.
+
+### Comparison between Directive definition and Component definition
+
+|                   | Directive            | Component       |
+|-------------------|----------------------|-----------------|
+| bindings          | No                   | Yes (binds to controller)      |
+| bindToController  | Yes (default: false) | No (use bindings instead)       |
+| compile function  | Yes                  | No       |
+| controller        | Yes                  | Yes      |
+| controllerAs      | Yes (default: false) | Yes (default: `$ctrl`)       |
+| link functions    | Yes                  | No       |
+| multiElement      | Yes                  | No       |
+| priority          | Yes                  | No       |
+| require           | Yes                  | Yes      |
+| restrict          | Yes                  | No (restricted to elements only)      |
+| scope             | Yes (default: false) | No (scope is always isolate)      |
+| template          | Yes                  | Yes, injectable      |
+| templateNamespace | Yes                  | No       |
+| templateUrl       | Yes                  | Yes, injectable       |
+| terminal          | Yes                  | No       |
+| transclude        | Yes (default: false) | Yes (default: false)      |
+
+
+## Component-based application architecture
+
+As already mentioned, the component helper makes it easier to create your application with
```

Comment:
I'd say something like: `...to architect your application in a component-based style.`
Or: `...to apply a component-based architecture to your application.`